account/manager.go

```diff
@@ -798,6 +801,150 @@ func (m *manager) resumeAccount(ctx context.Context, account *Account, // nolint
 	return nil
 }
 
+// isAccountCreation returns true if the given transaction label denotes an
+// account creation action.
+func isAccountCreation(txLabel string) bool {
+	return strings.HasPrefix(txLabel, " poold -- AccountCreation(")
+}
+
+// isAccountModification returns true if the given transaction label denotes an
+// account modification action.
+func isAccountModification(txLabel string) bool {
+	return strings.HasPrefix(txLabel, " poold -- AccountModification(")
+}
+
+// isAccountClosure returns true if the given transaction label denotes an
+// account closure action.
+func isAccountClosure(txLabel string) bool {
+	return strings.HasPrefix(txLabel, " poold -- AccountClosure(")
+}
+
+// These regular expressions (regex) match on transaction label flags.
+var reLabelAccountKey = regexp.MustCompile(`acct_key=(?P<acct_key>[0-9a-z]*)`)
+var reLabelExpiry = regexp.MustCompile(`expiry=(?P<expiry>(false|true))`)
+var reLabelDeposit = regexp.MustCompile(`deposit=(?P<deposit>(false|true))`)
+var reLabelIsClose = regexp.MustCompile(`is_close=(?P<is_close>(false|true))\)`)
```

Comment:
There are a few things I noticed while performing this label parsing on the frontend.

Here are the labels I see for each account action:
```
Create:
  poold -- AccountCreation(acct_key=028f4218ed13a50d9b6cc2afa947bba40e9336eaedbb10129ac1fda73a3d806252)
Deposit:
  poold -- AccountModification(acct_key=028f4218ed13a50d9b6cc2afa947bba40e9336eaedbb10129ac1fda73a3d806252, expiry=false, deposit=true, is_close=false)
Withdrawal:
  poold -- AccountModification(acct_key=028f4218ed13a50d9b6cc2afa947bba40e9336eaedbb10129ac1fda73a3d806252, expiry=false, deposit=false, is_close=false)
Renew:
  poold -- AccountModification(acct_key=028f4218ed13a50d9b6cc2afa947bba40e9336eaedbb10129ac1fda73a3d806252, expiry=false, deposit=false, is_close=false)
Close:
  poold -- AccountModification(acct_key=028f4218ed13a50d9b6cc2afa947bba40e9336eaedbb10129ac1fda73a3d806252, expiry=false, deposit=false, is_close=true)
```

1. The `expiry` value is never `true`. I'm not sure when this is ever set to `true`. I assumed it would be for account renewals but that isn't the case. Maybe a bug?
2. There is no `withdraw` value in the label.
3. Due to the above, it's not possible to distinguish between withdrawals and renewals from the labels alone.

Just figured I'd point out the issue's I'm currently running into. Maybe they can also be addressed in this PR.

Reply:
Yes, those are the issues which are holding up this PR. Thank you for the clear summary!

I think it should be possible to distinguish between withdraw and renew based on the transaction outputs. But I'm not certain. The transaction outputs that I'm currently looking at (when it comes to withdraw) seem to be corrupted. Oli suspected lnd versioning mismatches.

Could I ask you to post a complete withdraw transaction please (you've given the label only in your comment)? I wander if it might look very different from what I can produce.

Reply:
As you've mentioned, the account renew action does not have a flag of its own and doesn't touch the expiry flag pathway. See: https://github.com/lightninglabs/pool/blob/9280be1cfe939a8fb9bec0bf0eef05376a84cdd7/account/manager.go#L1319-L1322

Reply:
https://github.com/lightninglabs/pool/pull/396